django/db/models/fields/__init__.py

```diff
@@ -1241,11 +1241,11 @@ def _check_fix_default_value(self):
         if not timezone.is_naive(now):
             now = timezone.make_naive(now, timezone.utc)
         value = self.default
-        if isinstance(value, datetime.datetime):
+        if hasattr(value, 'tzinfo') and hasattr(value, 'date'):
```

Comment:
Not sure if changes to these check methods are required -- was there a discussion somewhere about it? At least tests are missing for these changes.